source/reduce/operand_to_dominating_id_reduction_opportunity_finder.cpp

```diff
@@ -91,7 +91,8 @@ void OperandToDominatingIdReductionOpportunityFinder::
             // constant.  It is thus not relevant to this pass.
             continue;
           }
-          // Sanity check that we don't get here if the argument is a constant.
+          // Coherence check: we should not get here if the argument is a
```

Comment:
I'd remove "Coherence check" altogether, and start with "We ..."

Reply:
Done - and actually I have moved the comment into the assertion as a '&& "..."' string.